src/centralstore/incremental/NodeBlock.cpp

```diff
@@ -0,0 +1,261 @@
+            }
+        } else if (!this->isDirected && (currentRelation->destination.address == this->addr)) {
+            if (currentRelation->source.address == withNode.addr) {
+                found = currentRelation;
+                break;
+            } else {
+                currentRelation = currentRelation->nextDestination();
+            }
+        } else {
+            throw "Exception: Unrelated relation block for " + std::to_string(this->addr) +
+                " found in relation block " + std::to_string(currentRelation->addr);
+        }
+    }
+
+    return found;
+}
+
+std::list<NodeBlock> NodeBlock::getEdges() {
+    std::list<NodeBlock> edges;
+    RelationBlock* currentRelation = this->getRelationHead();
+    while (currentRelation != NULL) {
+        NodeBlock* node = NULL;
+        if (currentRelation->source.address == this->addr) {
+            node = NodeBlock::get(currentRelation->destination.address);
+            currentRelation = currentRelation->nextSource();
+        } else if (currentRelation->destination.address == this->addr) {
+            node = NodeBlock::get(currentRelation->source.address);
+            currentRelation = currentRelation->nextDestination();
+        } else {
+            throw "Error: Unrecognized relation for " + std::to_string(this->addr) + " in relation block " +
+                std::to_string(currentRelation->addr);
+        }
+        if (!node) {
+            throw "Error creating node in the relation";
+        }
+        edges.push_back(*node);
+    }
+    return edges;
+}
+
+std::map<std::string, char*> NodeBlock::getAllProperties() {
+    std::map<std::string, char*> allProperties;
+    PropertyLink* current = this->getPropertyHead();
+    while (current) {
+        allProperties.insert({current->name, current->value});
+        PropertyLink* _temp = current->next();
+        delete current;  // To prevent memory leaks
+        current = _temp;
+    }
+    delete current;
+    return allProperties;
+}
+
+NodeBlock* NodeBlock::get(unsigned int blockAddress) {
+    NodeBlock* nodeBlockPointer = NULL;
+    NodeBlock::nodesDB->seekg(blockAddress);
+    unsigned int edgeRef;
+    unsigned int propRef;
+    char usageBlock;
+    char label[6];
+    std::string id;
+
+    if (!NodeBlock::nodesDB->get(usageBlock)) {
+        node_block_logger.error("Error while reading usage data from block " + std::to_string(blockAddress));
+    }
+
+    if (!NodeBlock::nodesDB->read(reinterpret_cast<char*>(&edgeRef), sizeof(unsigned int))) {
+        node_block_logger.error("Error while reading edge reference data from block " + std::to_string(blockAddress));
+    }
+
+    if (!NodeBlock::nodesDB->read(reinterpret_cast<char*>(&propRef), sizeof(unsigned int))) {
+        node_block_logger.error("Error while reading prop reference data from block " + std::to_string(blockAddress));
+    }
+
+    if (!NodeBlock::nodesDB->read(&label[0], 6)) {
```

Comment:
Can we replace 6 with the constant once defined?

Reply:
Fixed in https://github.com/miyurud/jasminegraph/pull/115/commits/4381d28b31e09cf40c82321c326bdb8aebbe55ce

Reply:
Done